great_expectations/core/batch.py

```diff
@@ -574,3 +587,180 @@ def head(self, n_rows=5, fetch_all=False):
+        raise ge_exceptions.GreatExpectationsTypeError(
+            f"the first parameter, datasource_name, must be a str, not {type(datasource_name)}"
+        )
+
+    if len([arg for arg in [batch_data, query, path] if arg is not None]) > 1:
+        raise ValueError("Must provide only one of batch_data, query, or path.")
+
+    if any(
+        [
+            batch_data is not None
+            and runtime_parameters
+            and "batch_data" in runtime_parameters,
+            query and runtime_parameters and "query" in runtime_parameters,
+            path and runtime_parameters and "path" in runtime_parameters,
+        ]
+    ):
+        raise ValueError(
+            "If batch_data, query, or path arguments are provided, the same keys cannot appear in the "
+            "runtime_parameters argument."
+        )
+
+    if batch_request:
+        # TODO: Raise a warning if any parameters besides batch_requests are specified
+        return batch_request
+
+    if any([batch_data is not None, query, path, runtime_parameters]):
+        runtime_parameters = runtime_parameters or {}
+        if batch_data is not None:
+            runtime_parameters["batch_data"] = batch_data
+        elif query is not None:
+            runtime_parameters["query"] = query
+        elif path is not None:
+            runtime_parameters["path"] = path
+
+        if batch_identifiers is None:
+            batch_identifiers = kwargs
+        else:
+            # Raise a warning if kwargs exist
+            pass
+
+        batch_request = RuntimeBatchRequest(
+            datasource_name=datasource_name,
+            data_connector_name=data_connector_name,
+            data_asset_name=data_asset_name,
+            runtime_parameters=runtime_parameters,
+            batch_identifiers=batch_identifiers,
+            batch_spec_passthrough=batch_spec_passthrough,
+        )
+    else:
+        if data_connector_query is None:
+            if batch_filter_parameters is not None and batch_identifiers is not None:
+                raise ValueError(
+                    'Must provide either "batch_filter_parameters" or "batch_identifiers", not both.'
+                )
+            elif batch_filter_parameters is None and batch_identifiers is not None:
+                logger.warning(
+                    'Attempting to build data_connector_query but "batch_identifiers" was provided '
+                    'instead of "batch_filter_parameters". The "batch_identifiers" key on '
+                    'data_connector_query has been renamed to "batch_filter_parameters". Please update '
+                    'your code. Falling back on provided "batch_identifiers".'
+                )
+                batch_filter_parameters = batch_identifiers
+            elif batch_filter_parameters is None and batch_identifiers is None:
+                batch_filter_parameters = kwargs
+            else:
+                # Raise a warning if kwargs exist
```

Comment:
If we don't intend to raise these warnings, should we just remove these clauses? On a side note, we should make sure the "War on Warnings" is differentiating between dependency warnings and GE emitted warnings.

Reply:
Discussed in a call with @NathanFarmer and resolved for now.